code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -527,22 +527,32 @@ func (fsh *StorageHandler) getReferencePath(ctx context.Context, r *http.Request
 	}
 	allocationID := allocationObj.ID
 
-	clientSign, _ := ctx.Value(constants.ContextKeyClientSignatureHeaderKey).(string)
-	valid, err := verifySignatureFromRequest(allocationTx, clientSign, allocationObj.OwnerPublicKey)
-	if !valid || err != nil {
-		errCh <- common.NewError("invalid_signature", "Invalid signature")
+	paths, err := pathsFromReq(r)
+	if err != nil {
+		errCh <- err
 		return
 	}
 
+	clientSign, _ := ctx.Value(constants.ContextKeyClientSignatureHeaderKey).(string)
+
 	clientID := ctx.Value(constants.ContextKeyClient).(string)
 	if clientID == "" {
 		errCh <- common.NewError("invalid_operation", "Please pass clientID in the header")
 		return
 	}
 
-	paths, err := pathsFromReq(r)
-	if err != nil {
-		errCh <- err
+	publicKey := allocationObj.OwnerPublicKey
+
+	// it is not owner, check if it is a colloborator
+	if allocationObj.OwnerID != clientID && len(clientID) > 0 {
+		if reference.IsCollaboratorInAllPaths(ctx, allocationID, paths, clientID) {
+			publicKey = ctx.Value(constants.ContextKeyClientKey).(string)
+		}
+	}
+
+	valid, err := verifySignatureFromRequest(allocationTx, clientSign, publicKey)
+	if !valid || err != nil {
+		errCh <- common.NewError("invalid_signature", "Invalid signature")
```

Comment:
Perhaps return error of `common.NewError("invalid_signature", "could not verify the allocation owner or colloborator") to make the error message more obvious.

Reply:
fixed